### PR TITLE
Hide overlapping labels in crossAxis

### PR DESCRIPTION
**Pull request by andy-lee-eng**

### CODE REVIEW

File: packages/perspective-viewer-d3fc/src/js/axis/crossAxis.js [hunk 1]
```diff
@@ -112,7 +112,34 @@ export const styleAxis = (chart, prefix, settings) => {
...
+    const previousRectangles = [];
+    s.each((d, i, nodes) => {
+        const tick = d3.select(nodes[i]);
+        const text = tick.select("text");
+        const textRect = text.node().getBBox();
+
+        const transformCoords = getTransformCoords(tick.attr("transform"));
+
+        const rect = {x: textRect.x + transformCoords[0], y: textRect.y + transformCoords[1], width: textRect.width, height: textRect.height};
+        const overlap = !!previousRectangles.find(r => rectanglesOverlap(r, rect));
```
Comment (Ro4052) [inline packages/perspective-viewer-d3fc/src/js/axis/crossAxis.js:138]: Just a question, what does the double `!` provide?

**Review by Ro4052 — APPROVED** [review]

**Review by mhooper-scottlogic — CHANGES_REQUESTED** [review]
It works great for Bar/Column charts with ordinal values but seems to cause issues for other series types.

E.g.
http://localhost:3000/MetOffice/MetOffice.html

Before:
image

After
image

http://localhost:3000/Currency/currency.html
Before:
image

After:
image

File: packages/perspective-viewer-d3fc/src/js/axis/crossAxis.js (hunk 1 above)
Comment (mhooper-scottlogic) [reply packages/perspective-viewer-d3fc/src/js/axis/crossAxis.js:138]: I think it returns a value's truthy value:
https://medium.com/@pddivine/javascript-bang-bang-i-shot-you-down-use-of-double-bangs-in-javascript-7c9d94446054
In this context find will return the first overlap, but we don't need to know the details of the overlap, just that it exists. so !! will return true if any values come back and false if there are no results (e.g. undefined).

I think we could also do:
const overlap = previousRectangles.some(r => rectanglesOverlap(r, rect));

Comment (Ro4052) [reply packages/perspective-viewer-d3fc/src/js/axis/crossAxis.js:138]: I guess it's casting into a boolean, so it's a little safer (and makes a little more sense) than keeping the value as an object.

**Review by mhooper-scottlogic — APPROVED** [review]

**Pull request merged**